src/Gui/MainWindow.cpp

```diff
@@ -150,11 +150,92 @@ class CustomMessageEvent : public QEvent
     int _timeout;
 };
 
+class DimensionWidget : public QPushButton, WindowParameter
```

Comment:
Maybe some documentation could be added before the class to explain briefly what is this widget.

Reply:
Yes, will add before merging.

Reply:
Done